backend/src/main/java/com/woowacourse/levellog/application/TeamService.java

```diff
@@ -117,9 +123,33 @@ private Long getLevellog(final Participant participant) {
         return levellog.getId();
     }
 
-    public void update(final Long id, final TeamUpdateRequest request) {
-        final Team team = teamRepository.findById(id)
-                .orElseThrow(TeamNotFoundException::new);
+    public void update(final Long id, final TeamUpdateRequest request, final Long memberId) {
+        final Team team = findTeam(id);
+        final List<Participant> participants = participantRepository.findByTeam(team);
+        final Long hostId = getHostId(participants);
+
+        if (!memberId.equals(hostId)) {
+            throw new HostUnauthorizedException();
+        }
+
         team.update(request.getTitle(), request.getPlace(), request.getStartAt());
     }
+
+    public void deleteById(final Long id, final Long memberId) {
+        final Team team = findTeam(id);
+        final List<Participant> participants = participantRepository.findByTeam(team);
+        final Long hostId = getHostId(participants);
```

Comment:
(안 중요하긴 함)

`getHostId(participants)`메서드르 사용하는 모든 곳에서 `final List<Participant> participants = participantRepository.findByTeam(team);`가 선행되는 것 같고 앞으로도 그럴 것 같아요. 참가자를 찾는 로직을 `getHostId()`메서드에 포함시키고 아래처럼 수정해도 좋을 것 같아요

```suggestion
        final Long hostId = getHostIdByTeam(team);
```

Reply:
팀 CRUD 많이 빡셌을텐데 수고하셨어요 ! 👍 👍 👍

Reply:
코드 리팩토링을 위한 좋은 의견 감사해요 알린!!
현재 `getTeam(final Team team)` 메서드를 보면, participants가 `getHostId()` 뿐만 아니라 참가자DTO로 만들어주는 `getParticipantsResponse()`메서드에서도 필요로 하여 인자로 받고 있습니다. 
그렇기 때문에 똑같은 데이터를 두 번 조회를 하지 않도록 `participantRepository.findByTeam(team);`를 분리하였습니다. 
이에 대한 알린 생각은 어떤가요??👀